calico_rkt/calico_rkt.py

```diff
@@ -20,381 +20,474 @@
+
+        :param endpoint
+        :return Calico endpoint object
+        """
+        netns_path = os.path.abspath(os.path.join(os.getcwd(), self.netns))
+        endpoint.mac = endpoint.provision_veth(Namespace(netns_path),
+                                               self.interface)
+        self._client.set_endpoint(endpoint)
+        _log.info("Provisioned veth for endpoint using netns path %s on "
+                  "interface %s", netns_path, self.interface)
+        return endpoint
+
+    def _set_profile_on_endpoint(self, endpoint, profile_name):
+        """Sets given profile_name on given endpoint.
+
+        If profile for given profile_name does not exist, create a profile.
+
+        :param endpoint
+        :param profile_name
+        :return: None.
+        """
+        if not self._client.profile_exists(profile_name):
+            _log.info("Creating new profile %s.", profile_name)
+            self._client.create_profile(profile_name)
+
+        _log.info("Setting profile %s on endpoint.", profile_name)
+        self._client.set_profiles_on_endpoint(profile_names=[profile_name],
+                                              endpoint_id=endpoint.endpoint_id)
 
-    # Remove the container from the datastore.
-    datastore_client.remove_workload(hostname=hostname,
-                                     orchestrator_id=orchestrator_id,
-                                     workload_id=container_id)
+    def _remove_profile(self, profile_name):
+        """Remove the profile from datastore if it has no endpoints attached.
+
+        :param profile_name: Name of profile to remove.
+        :return: None.
+        """
+        profile_members = self._client.get_profile_members(profile_name)
+        if self._client.profile_exists(profile_name) and len(profile_members) < 1:
+            try:
+                _log.info("Profile %s has no members, removing from datastore.",
+                          profile_name)
+                self._client.remove_profile(profile_name)
+            except KeyError:
+                _log.error("Cannot remove profile %s: Profile not found.",
+                           profile_name)
+                sys.exit(1)
+
+    def _get_endpoint(self):
+        """Gets endpoint matching the container_id.
+
+        Exits gracefully if no endpoint is found.
+        Exits with an error if multiple endpoints are found.
+
+        :param container_id:
+        :return: Calico endpoint object
+        """
+        try:
+            _log.info("Retrieving endpoint that matches container ID %s",
+                      self.container_id)
+            ep = self._client.get_endpoint(hostname=HOSTNAME,
+                                           orchestrator_id=ORCHESTRATOR_ID,
+                                           workload_id=self.container_id)
+        except KeyError:
+            _log.warning("No endpoint found matching ID %s", self.container_id)
+            sys.exit(0)
```

Comment:
I prefer we follow the format of the exec plugin - if the endpoint isn't found, log a debug log and return None.  The calling code might not want to exit if the endpoint doesn't exist, so it can make that decision on its own.

e.g
```
# Check if already exists.
endpoint = self._get_endpoint()
if endpoint:
  _log.error("Endpoint already exists!")
  return

_log.info("Endpoint does not exist, creating")
```